### ISSUE TITLE
Http.POST not finding the specified URL.

### ISSUE THREAD

**Issue by JimJim87** [Starcounter/Starcounter#3988]
When doing a Http.POST to my own asp.net web-api I get a Http 400 Bad Request response. When I try the exact same url using postman I get a 200 OK.

I have successfully made posts to requestbin.

```
var dict = new Dictionary<string, string>();
dict.Add("Content-Type", "application/x-www-form-urlencoded");
Response testResponse;

Http.POST("http://localhost:61124/api/startconfigurator", "teststring", dict, (Response resp) => {
    testResponse = resp;
});
Http.POST(@"http://requestb.in/1640fhj1", "teststring", dict, (Response resp) =>
{
    test = resp;
});
```
Requestbin post:
![image](https://cloud.githubusercontent.com/assets/20333891/20758625/b0a56c42-b71a-11e6-9aaa-9be5a2f0f20e.png)

400 Response from own API:
![image](https://cloud.githubusercontent.com/assets/20333891/20758336/c111a9a2-b719-11e6-92e7-700b10bd6771.png)

Postman to own API:
![image](https://cloud.githubusercontent.com/assets/20333891/20757978/a8b8113a-b718-11e6-9667-207743c982ce.png)

Am I missing something simple here?

Regards

**Comment by miyconst** [comment]
@JimJim87 please try to use native .NET features instead of `Starcounter.Http` class and see whether it works.

**Comment by miyconst** [comment]
cc @alemoi

**Comment by JimJim87** [comment]
@miyconst I tried a solution found on http://stackoverflow.com/questions/4088625/net-simplest-way-to-send-post-with-data-and-read-response (it's in the accepted answer) and it worked.

I landed in my Post handler in my ASP.NET web-api. Not sure how to convert the response but that is another problem.

```
            using (WebClient client = new WebClient())
            {

                byte[] response =
                client.UploadValues("http://localhost:61124/api/startconfigurator", new NameValueCollection()
                {
                    { "home", "Cosby" },
                    { "favorite+flavor", "flies" }
                });

                string result = System.Text.Encoding.UTF8.GetString(response);
            }
```

Should I use something like that for now?

**Comment by miyconst** [comment]
Yes, please use `WebClient` as it is a native feature, and I am not sure of how long the `Starcounter.Http` will be supported.

**Comment by alemoi** [comment]
@miyconst I agree with your suggestion. Not sure why Asp.net returns 400 but in this case we need to inspect both request and response using Fiddler, for example.
